templates/apiary-kms-key-policy.json

```diff
@@ -0,0 +1,64 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+%{if client_roles != ""}
+    {
+      "Sid": "Allow use of the key",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": ["${client_roles}"]
+      },
+      "Action": [
+        "kms:Encrypt",
+        "kms:Decrypt",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:DescribeKey"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "Allow attachment of persistent resources",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": ["${client_roles}"]
```

Comment:
Will this work for cross-account roles?  I imagine soon after this rolls out, someone will have a use-case to access the bucket cross-account.

Reply:
in theory yes, but we may not have usecases to allow cross account access to KMS encrypted buckets as we will have little control on who is access the data one cross accounts access is allowed.